CRM/Core/BAO/Dashboard.php

```diff
@@ -40,37 +42,63 @@ public static function create($params) {
    *
    * @return array
    *   array of dashlets
+   * @throws \API_Exception
    */
-  public static function getContactDashlets() {
-    if (!isset(Civi::$statics[__CLASS__][__FUNCTION__])) {
-      Civi::$statics[__CLASS__][__FUNCTION__] = [];
+  public static function getContactDashlets(): array {
+    $cid = CRM_Core_Session::getLoggedInContactID();
+    if ($cid && !isset(Civi::$statics[__CLASS__][__FUNCTION__][$cid])) {
+      Civi::$statics[__CLASS__][__FUNCTION__][$cid] = [];
+      // If empty, then initialize default dashlets for this user.
+      if (0 === DashboardContact::get(FALSE)->selectRowCount()->addWhere('contact_id', '=', $cid)->execute()->count()) {
+        self::initializeDashlets();
+      }
+      $contactDashboards = (array) DashboardContact::get(FALSE)
+        ->addSelect('column_no', 'is_active', 'dashboard_id', 'weight', 'contact_id')
+        ->addWhere('contact_id', '=', $cid)
+        ->addOrderBy('weight')
+        ->execute()->indexBy('dashboard_id');
+
       $params = [
         'select' => ['*', 'dashboard_contact.*'],
-        'join' => [
-          ['DashboardContact AS dashboard_contact', FALSE, ['dashboard_contact.contact_id', '=', CRM_Core_Session::getLoggedInContactID()]],
-        ],
         'where' => [
           ['domain_id', '=', 'current_domain'],
         ],
-        'orderBy' => ['dashboard_contact.weight' => 'ASC'],
       ];
 
       // Get Dashboard + any joined DashboardContact records.
-      $results = civicrm_api4('Dashboard', 'get', $params);
-
-      // If empty, then initialize default dashlets for this user.
-      if (!array_filter($results->column('dashboard_contact.id'))) {
-        self::initializeDashlets();
-      }
-      $results = civicrm_api4('Dashboard', 'get', $params);
-
+      $results = (array) civicrm_api4('Dashboard', 'get', $params);
       foreach ($results as $item) {
+        $item['dashboard_contact.id'] = $contactDashboards[$item['id']]['contact_id'] ?? NULL;
```

Comment:
Ah - this line - @sunilpawar do you want to try changing this to 
```
$item['dashboard_contact.contact_id'] = $contactDashboards[$item['id']]['contact_id'] ?? NULL;
$item['dashboard_contact.id'] = $contactDashboards[$item['id']]['id'] ?? NULL;
```

If that works I'll push it up